### PR TITLE
Upgrade to operator-sdk & api version 0.18

### PR DESCRIPTION
**Pull request by bdunne**
- Follow guidance from https://kubernetes.io/docs/reference/using-api/deprecation-guide/#customresourcedefinition-v122
  - v1beta1 CRD is removed in Kubernetes 1.22, move to v1
- Remove a bunch of replaces that don't seem to be necessary
- HACK: Use forked version of k8s.io/api v0.18 with some components of v0.19 added in https://github.com/bdunne/kubernetes-api/commit/707adf1e9ef5473bccae78b63fdb32861dd8200d
- Move to networking/v1 Ingress

Testing
- [X] Kubernetes 1.22 (OpenShift 4.9.5)
- [x] Kubernetes 1.19.16 (OpenShift 4.6.52)

Fixes: https://github.com/ManageIQ/manageiq-pods/issues/777

EDIT: ~~moved back to WIP because the vendor directory didn't work as expected.~~ Used a fork instead

### CODE REVIEW

File: manageiq-operator/go.sum [hunk 1]
```diff
@@ -98,6 +100,8 @@ github.com/aws/aws-sdk-go v1.15.11/go.mod h1:mFuSZ37Z9YOHbQEwBWztmVzqXrEkub65tZo
 github.com/aws/aws-sdk-go v1.17.7/go.mod h1:KmX6BPdI08NWTb3/sm4ZGu5ShLoqVDhKgpiN924inxo=
 github.com/aws/aws-sdk-go v1.25.48/go.mod h1:KmX6BPdI08NWTb3/sm4ZGu5ShLoqVDhKgpiN924inxo=
 github.com/baiyubin/aliyun-sts-go-sdk v0.0.0-20180326062324-cfa1a18b161f/go.mod h1:AuiFmCCPBSrqvVMvuqFuk0qogytodnVFVSN5CeJB8Gc=
+github.com/bdunne/api v0.0.0-20220110152537-707adf1e9ef5 h1:Hh720QgbNiOZooChZQIsTFX5SSKW2HFycrEXoz3e/LU=
+github.com/bdunne/api v0.0.0-20220110152537-707adf1e9ef5/go.mod h1:SJCWI7OLzhZSvbY7U8zwNl9UA4o1fizoug34OV/2r78=
```
Comment (chessbyte) [inline manageiq-operator/go.sum:104]: are these personal repos right?

Comment (bdunne) [reply manageiq-operator/go.sum:104]: yes, it is.

File: manageiq-operator/go.mod [hunk 2]
```diff
@@ -4,41 +4,18 @@ go 1.14
 
 require (
 	github.com/openshift/api v0.0.0-20200205133042-34f0ec8dab87
-	github.com/operator-framework/operator-sdk v0.17.2
+	github.com/operator-framework/operator-sdk v0.18.0
```
Comment (Fryguy) [inline manageiq-operator/go.mod:7]: I expected 0.18.2 here.

**Review by Fryguy — APPROVED** [review]

File: manageiq-operator/go.mod [hunk 3]
```diff
@@ -4,41 +4,18 @@ go 1.14
...
-	k8s.io/kube-aggregator => k8s.io/kube-aggregator v0.0.0-20191016112429-9587704a8ad4
-	k8s.io/kube-controller-manager => k8s.io/kube-controller-manager v0.0.0-20191016114939-2b2b218dc1df
-	k8s.io/kube-proxy => k8s.io/kube-proxy v0.0.0-20191016114407-2e83b6f20229
-	k8s.io/kube-scheduler => k8s.io/kube-scheduler v0.0.0-20191016114748-65049c67a58b
-	k8s.io/kubectl => k8s.io/kubectl v0.0.0-20191016120415-2ed914427d51
-	k8s.io/kubelet => k8s.io/kubelet v0.0.0-20191016114556-7841ed97f1b2
-	k8s.io/legacy-cloud-providers => k8s.io/legacy-cloud-providers v0.0.0-20191016115753-cf0698c3a16b
-	k8s.io/metrics => k8s.io/metrics v0.0.0-20191016113814-3b1a734dba6e
-	k8s.io/sample-apiserver => k8s.io/sample-apiserver v0.0.0-20191016112829-06bb3c9d77c9
+	k8s.io/api => github.com/bdunne/kubernetes-api v0.0.0-20220110152537-707adf1e9ef5
```
Comment (Fryguy) [inline manageiq-operator/go.mod:19]: Can you add a HACK comment here?  Something like:

```suggestion
        // Temporary fork of the kubernetes-api to include some 0.19 changes until we can get to 0.19
	k8s.io/api => github.com/bdunne/kubernetes-api v0.0.0-20220110152537-707adf1e9ef5
```

**Comment by Fryguy** [comment]
Instead of pointing to a fork, would it be possible to reopen the v1 package and insert those types with some local go file in this project?

For the `AddKnownTypes`, perhaps we can just call it directly?

File: manageiq-operator/go.sum (hunk 1 above)
Comment (Fryguy) [reply manageiq-operator/go.sum]: Moved to a ManageIQ fork.

**Comment by Fryguy** [comment]
Backported to `morphy` via #784

**Pull request merged**